### PR TITLE
Handle kill signaling and improve StartUp CLI user experience

### PR DESCRIPTION
**Pull request by ja0n**
- [X] SIGTERM handling: graceful shutdown on kill signal
- [X] CTRL-C handling
- [X] Exit Confirmation
- [X] help command
- [X] pid command
- [X] daemon mode param `-daemon`
- [x] Update Docs

### CODE REVIEW

**Comment by sven-n** [comment]
Question: Why is the handling of Ctrl+C needed?
Hint: When entering 'exit' or pressing Ctrl+C the application is not exiting anymore :-(

**Comment by ja0n** [comment]
> Question: Why is the handling of Ctrl+C needed?
> Hint: When entering 'exit' or pressing Ctrl+C the application is not exiting anymore :-( It shuts down, but then the process gets stuck.

The Ctrl+C handling improves developer experience
I build and run the project on CLI, so the Ctrl+C feels natural :grin: 

So about the stuck... that was the original issue, it isn't working for me under `master`. I haven't tried on Windows tho
My last commit fixed it for me (Linux). Can you test again?

**Comment by ja0n** [comment]
Now I'm trying to handle the SIGTERM (kill) signal, it isn't working for me yet.
Oh, there's the pipeline error to fix too

File: src/Startup/Program.cs [hunk 1]
```diff
@@ -101,11 +101,9 @@ public static void Main(string[] args)
 
             using (new Program(args))
             {
-                bool exit = false;
+                var exit = false;
```
Comment (ja0n) [inline src/Startup/Program.cs:104]: is that a code style rule? just curious

Comment (sven-n) [reply src/Startup/Program.cs:104]: no, but you can easily spot that ```exit``` is a bool ;-)

**Comment by sven-n** [comment]
> My last commit fixed it for me (Linux). Can you test again?

Yes, works on windows, too. I just noticed one issue, that when you press Ctrl+C, Console.ReadLine() returns with null, but the Event is handled afterwards… so it waits for another ReadLine() :-/

**Comment by sven-n** [comment]
> Yes, works on windows, too. I just noticed one issue, that when you press Ctrl+C, Console.ReadLine() returns with null, but the Event is handled afterwards… so it waits for another ReadLine() :-/

Just fixed that :)

**Comment by ja0n** [comment]
Well, I had fixed it too. I also managed to handle SIGTERM signaling :)
Now I can run it as a daemon service and gracefully shutdown the server
I'll rebase and resolve conflicts tomorrow so you can review it! thanks

File: src/Startup/Program.cs [hunk 2]
```diff
@@ -94,35 +95,39 @@ public Program(string[] args)
...
 
-            using (new Program(args))
+            using var program = new Program(args);
+            using var exitCts = new CancellationTokenSource();
+            var exitToken = exitCts.Token;
+            void OnCancelKeyPress(object sender, ConsoleCancelEventArgs e)
             {
-                var exit = false;
+                exitCts.Cancel();
+                Console.CancelKeyPress -= OnCancelKeyPress;
```
Comment (ja0n) [inline src/Startup/Program.cs:109]: is that neccessary?

File: src/Startup/TextReaderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        /// <summary>
+        /// Reads the line asynchronously and considers the <see cref="CancellationToken"/>.
+        /// </summary>
+        /// <param name="reader">The reader.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The read line or <c>null</c>.</returns>
+        public static async Task<string> ReadLineAsync(this TextReader reader, CancellationToken cancellationToken)
+        {
+            using var taskSource = new CancellationTokenTaskSource<string>(cancellationToken);
+            var result = await await Task.WhenAny(taskSource.Task, reader.ReadLineAsync());
```
Comment (ja0n) [inline src/Startup/TextReaderExtensions.cs:26]: Very smart! It fixes a problem I couldn't solve
I'd just split this into two lines to make it easier to read :grin:

File: src/Startup/Program.cs (hunk 2 above)
Comment (sven-n) [reply src/Startup/Program.cs:109]: It’s just to be sure that CancellationTokenSource.Cancel() isn’t called after it has been disposed. When the application wasn’t exited, I noticed that pressing Ctrl+C another time caused an exception and crash.

Comment (ja0n) [reply src/Startup/Program.cs]: oh, i just saw this crash. should keep this then. thnx

Comment (ja0n) [reply src/Startup/Program.cs]: this exception?
```
Unhandled exception. System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at MUnique.OpenMU.Startup.TextReaderExtensions.ReadLineAsync(TextReader reader, CancellationToken cancellationToken) in /home/ja0n/Projects/OpenMU/src/Startup/TextReaderExtensions.cs:line 26
   at MUnique.OpenMU.Startup.Program.Main(String[] args) in /home/ja0n/Projects/OpenMU/src/Startup/Program.cs:line 137
   at MUnique.OpenMU.Startup.Program.<Main>(String[] args)
```

Comment (ja0n) [reply src/Startup/Program.cs]: this exception?
```
Unhandled exception. System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at MUnique.OpenMU.Startup.TextReaderExtensions.ReadLineAsync(TextReader reader, CancellationToken cancellationToken)
```

**Comment by ja0n** [comment]
In my last commits I added:
- [SIGTERM](https://pt.wikipedia.org/wiki/SIGTERM) handling: graceful shutdown on kill signal
- Exit Confirmation
- help command
- pid command

**Pull request merged**